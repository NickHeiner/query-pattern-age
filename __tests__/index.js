const execa = require('execa');
const packageJson = require('../package.json');
const path = require('path');

const binPath = path.resolve(__dirname, '..', packageJson.bin['query-pattern-age']);
const astArgs = ['--astSelector', 'CallExpression[callee.object.name=console][callee.property.name=log]'];
const baseArgs = [
  '--paths', path.resolve(__dirname, '..', '__fixtures__'), 
  ...astArgs
];

describe('query-pattern-age', () => {
  it('raw', async () => {
    const {stdout} = await execa(binPath, [...baseArgs, '--format', 'raw']);
    expect(JSON.parse(stdout)).toMatchSnapshot();
  });
  it('pretty', async () => {
    const {stdout} = await execa(binPath, [...baseArgs, '--format', 'pretty']);
    expect(stdout).toMatchSnapshot();
  });
  it('only one file', async () => {
    const {stdout} = await execa(binPath, [
      ...astArgs, 
      '--format', 'raw', 
      '--paths', path.resolve(__dirname, '..', '__fixtures__', 'also-contains-pattern.js')
    ]);
    expect(JSON.parse(stdout)).toMatchSnapshot();
  });
});