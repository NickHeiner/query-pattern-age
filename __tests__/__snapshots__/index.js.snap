// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query-pattern-age only one file 1`] = `
Array [
  Object {
    "author": "Geralt",
    "commands": Array [
      "git blame /Users/nheiner/code/query-pattern-age/__fixtures__/also-contains-pattern.js -L 1,1 --porcelain",
    ],
    "count": 1,
    "files": Array [
      "/Users/nheiner/code/query-pattern-age/__fixtures__/also-contains-pattern.js",
    ],
    "hash": "b08f3fb9ec58b81941c22dc9efb1b9034ac7e4c7",
    "timestampS": 1578441779,
  },
]
`;

exports[`query-pattern-age pretty 1`] = `
"[90m┌─────────────[39m[90m┬───────────[39m[90m┬──────────────────[39m[90m┬─────────────────────────────────────────────────────────────────────────────[39m[90m┬──────────────────────────────────────────┐[39m
[90m│[39m[31m Date        [39m[90m│[39m[31m Committer [39m[90m│[39m[31m Occurrence Count [39m[90m│[39m[31m Files                                                                       [39m[90m│[39m[31m Hash                                     [39m[90m│[39m
[90m├─────────────[39m[90m┼───────────[39m[90m┼──────────────────[39m[90m┼─────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────────────────┤[39m
[90m│[39m 2020/Jan/07 [90m│[39m Geralt    [90m│[39m 2                [90m│[39m /Users/nheiner/code/query-pattern-age/__fixtures__/also-contains-pattern.js [90m│[39m b08f3fb9ec58b81941c22dc9efb1b9034ac7e4c7 [90m│[39m
[90m│[39m             [90m│[39m           [90m│[39m                  [90m│[39m /Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js      [90m│[39m                                          [90m│[39m
[90m├─────────────[39m[90m┼───────────[39m[90m┼──────────────────[39m[90m┼─────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────────────────┤[39m
[90m│[39m 2020/Jan/07 [90m│[39m Ciri      [90m│[39m 1                [90m│[39m /Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js      [90m│[39m a9ba6f6f0b924f711e1a4e588154362d06dedab8 [90m│[39m
[90m└─────────────[39m[90m┴───────────[39m[90m┴──────────────────[39m[90m┴─────────────────────────────────────────────────────────────────────────────[39m[90m┴──────────────────────────────────────────┘[39m"
`;

exports[`query-pattern-age raw 1`] = `
Array [
  Object {
    "author": "Geralt",
    "commands": Array [
      "git blame /Users/nheiner/code/query-pattern-age/__fixtures__/also-contains-pattern.js -L 1,1 --porcelain",
      "git blame /Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js -L 1,1 -L 3,3 --porcelain",
    ],
    "count": 2,
    "files": Array [
      "/Users/nheiner/code/query-pattern-age/__fixtures__/also-contains-pattern.js",
      "/Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js",
    ],
    "hash": "b08f3fb9ec58b81941c22dc9efb1b9034ac7e4c7",
    "timestampS": 1578441779,
  },
  Object {
    "author": "Ciri",
    "commands": Array [
      "git blame /Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js -L 1,1 -L 3,3 --porcelain",
    ],
    "count": 1,
    "files": Array [
      "/Users/nheiner/code/query-pattern-age/__fixtures__/contains-pattern.js",
    ],
    "hash": "a9ba6f6f0b924f711e1a4e588154362d06dedab8",
    "timestampS": 1578443564,
  },
]
`;
